apps/examples/security_test/hal/hal_keymgr_test.c

```diff
@@ -0,0 +1,175 @@
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#include <tinyara/security_hal.h>
+
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <pthread.h>
+
+#include <stress_tool/st_perf.h>
+#include "hal_test_utils.h"
+#include "hal_test.h"
+
+/*
+ * Desc: Set key
+ */
+static hal_data g_aes_key_in;
+TEST_SETUP(set_key)
+{
+	ST_START_TEST;
+
+	ST_EXPECT_EQ(0, hal_test_malloc_buffer(&g_aes_key_in, HAL_KEYMGR_TEST_MEM_SIZE));
+	g_aes_key_in.data_len = HAL_TEST_KEY_LEN;
+	memset(g_aes_key_in.data, 0xa5, HAL_TEST_KEY_LEN);
+
+	ST_END_TEST;
+}
+
+TEST_TEARDOWN(set_key)
+{
+	ST_START_TEST;
+	ST_EXPECT_EQ2(HAL_SUCCESS, HAL_NOT_SUPPORTED, hal_remove_key(HAL_KEY_AES_256, HAL_TEST_KEY_SLOT));
+	hal_test_free_buffer(&g_aes_key_in);
+	ST_END_TEST;
+}
+
+TEST_F(set_key)
+{
+	ST_START_TEST;
+	ST_EXPECT_EQ2(HAL_SUCCESS, HAL_NOT_SUPPORTED, hal_set_key(HAL_KEY_AES_256, HAL_TEST_KEY_SLOT, &g_aes_key_in, NULL));
+	ST_END_TEST;
+}
+
+/*
+ * Desc: Get key
+ * Refered https://developer.artik.io/documentation/security-api/see-authentication-test_8c-example.html
+ */
+static hal_data g_aes_key_out;
+TEST_SETUP(get_key)
+{
+	ST_START_TEST;
+
+	ST_EXPECT_EQ(0, hal_test_malloc_buffer(&g_aes_key_in, HAL_KEYMGR_TEST_MEM_SIZE));
+	g_aes_key_in.data_len = HAL_TEST_KEY_LEN;
+	memset(g_aes_key_in.data, 0xa5, HAL_TEST_KEY_LEN);
+
+	ST_EXPECT_EQ(0, hal_test_malloc_buffer(&g_aes_key_out, HAL_KEYMGR_TEST_MEM_SIZE));
+
+	ST_EXPECT_EQ2(HAL_SUCCESS, HAL_NOT_SUPPORTED, hal_set_key(HAL_KEY_AES_256, HAL_TEST_KEY_SLOT, &g_aes_key_in, NULL));
+
+	ST_END_TEST;
+}
+
+TEST_TEARDOWN(get_key)
+{
+	ST_START_TEST;
+
+	hal_test_free_buffer(&g_aes_key_in);
+	hal_test_free_buffer(&g_aes_key_out);
+
+	ST_EXPECT_EQ2(HAL_SUCCESS, HAL_NOT_SUPPORTED, hal_remove_key(HAL_KEY_AES_256, HAL_TEST_KEY_SLOT));
+
+	ST_END_TEST;
+}
+
+TEST_F(get_key)
+{
+	ST_START_TEST;
+
+	ST_EXPECT_EQ2(HAL_SUCCESS, HAL_NOT_SUPPORTED, hal_get_key(HAL_KEY_AES_256, HAL_TEST_KEY_SLOT, &g_aes_key_out));
+
+	ST_END_TEST;
+}
+
+/*
+ * Desc: Remove key in Secure Storage
+ */
+TEST_SETUP(remove_key)
+{
+	ST_START_TEST;
+
+	ST_EXPECT_EQ(0, hal_test_malloc_buffer(&g_aes_key_in, HAL_KEYMGR_TEST_MEM_SIZE));
+	g_aes_key_in.data_len = HAL_TEST_KEY_LEN;
+	memset(g_aes_key_in.data, 0xa5, HAL_TEST_KEY_LEN);
+
+	ST_EXPECT_EQ2(HAL_SUCCESS, HAL_NOT_SUPPORTED, hal_set_key(HAL_KEY_AES_256, HAL_TEST_KEY_SLOT, &g_aes_key_in, NULL));
+
+	ST_END_TEST;
+}
+
+TEST_TEARDOWN(remove_key)
+{
+	ST_START_TEST;
+
+	hal_test_free_buffer(&g_aes_key_in);
+
+	ST_END_TEST;
+}
+
+TEST_F(remove_key)
+{
+	ST_START_TEST;
+	ST_EXPECT_EQ2(HAL_SUCCESS, HAL_NOT_SUPPORTED, hal_remove_key(HAL_KEY_AES_256, HAL_TEST_KEY_SLOT));
+	ST_END_TEST;
+}
+
+/*
+ * Desc: Generate key
+ * Refered https://developer.artik.io/documentation/security-api/see-authentication-test_8c-example.html
+ */
+TEST_SETUP(generate_key)
+{
+	ST_START_TEST;
+
```

Comment:
same is this SETUP required here ?

Reply:
It should be defined, following the rule of stress tool. Thanks for the comment.